mono/mini/mini-arm.c

```diff
@@ -5909,9 +6007,16 @@ mono_arch_register_lowlevel_calls (void)
 	mono_register_jit_icall (mono_arm_throw_exception_by_token, "mono_arm_throw_exception_by_token", mono_create_icall_signature ("void"), TRUE);
 
 #ifndef MONO_CROSS_COMPILE
-#ifdef HAVE_AEABI_READ_TP
-	mono_register_jit_icall (__aeabi_read_tp, "__aeabi_read_tp", mono_create_icall_signature ("void"), TRUE);
-#endif
+	if (mono_arm_have_tls_get ()) {
+		if (mono_arm_have_fast_tls ()) {
+			mono_register_jit_icall (mono_fast_get_tls_key, "mono_get_tls_key", mono_create_icall_signature ("ptr ptr"), TRUE);
+			mono_register_jit_icall (mono_fast_set_tls_key, "mono_set_tls_key", mono_create_icall_signature ("void ptr ptr"), TRUE);
+		} else {
+			g_warning ("No fast tls on device. Using fallbacks.");
```

Comment:
We should avoid printing stuff unless absolutely necessary. Perhaps put this behind a log mask or something like that.